File: Data Structures & Algorithms/daily-temperatures/submission-2.cpp
class Solution {
public:
    vector<int> dailyTemperatures(const vector<int>& temperatures) {
        int n = temperatures.size();
        vector<int> res(n, 0);
        stack<int> stk{};
        for (int i{}; i < n; ++i) {
            int temp = temperatures[i];
            while (!stk.empty() && temperatures[stk.top()] < temp) {
                int day = stk.top(); stk.pop();
                int diff = i - day;
                res[day] = diff;
            }
            stk.push(i);
        }
        return res;
    }
};

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for c in tokens:
            if c == '+':
                stack.append(stack.pop() + stack.pop())
            elif c == '-':
                 a, b = stack.pop(), stack.pop()
                 stack.append(b - a)
            elif c == '*':
                 stack.append(stack.pop() * stack.pop())
            elif c == '/':
                 num1 = stack.pop()
                 num2 = stack.pop()
                
                 stack.append(int(float(num2) / num1))
            else:
                stack.append(int(c))
        return stack[0]